src/sources/kubernetes_logs/mod.rs

```diff
@@ -228,6 +234,13 @@ impl Source {
                 "unable to convert glob_minimum_cooldown_ms from usize to u64 without data loss",
             ));
 
+        let delay_deletion = Duration::from_millis(
+            config
+                .delay_deletion_ms
+                .try_into()
+                .expect("unable to convert delay_deletion_ms from usize to u64 without data loss"),
```

Comment:
I might change the `expect` here to something like: `we only support 64-bit architectures`.

Alternatively, could we just make the configuration option take a u64?

Reply:
I just ended up copying the existing implementation from the glob option, I can update both or leave as is - but seems weird to just do that for one 🤔

Reply:
I think that should probably just be a `u64` too. Others may have more informed opinions, but my general understanding is that `usize` is most useful when you need a value that is directly related to memory size or indexing; otherwise use a specifically sized type.

Reply:
I'll update allthethings

Reply:
It was noted that all of the other integer types in the configuration are `usize` so I think it's fine to leave this one for consistency.